### ISSUE TITLE
[Optimizer Proposal] NEAT: conflict-aware optimization via directional gradient correction

### ISSUE THREAD

**Issue by ItCodinTime** [NVIDIA-NeMo/Emerging-Optimizers#295]
## Is your feature request related to a problem?

Many emerging optimizers improve scaling, conditioning, curvature approximation, or optimizer memory.

A complementary problem is directional gradient conflict.

In some training regimes, the current gradient can directly oppose a stateful optimization signal such as momentum. Standard first-order optimizers generally continue accumulating that gradient without explicitly measuring the disagreement.

I would like to propose NEAT, Nash-Equilibrium Adaptive Training, as an experimental conflict-aware optimizer for Emerging-Optimizers.

Reference implementation:

https://github.com/ItCodinTime/NEAT

NEAT is also listed in the official Keras ecosystem:

https://keras.io/getting_started/ecosystem/

## Proposed method

For current gradient `g_t` and opponent proxy `o_t`, NEAT measures conflict using negative cosine similarity.

```python
conflict = relu(-cos(g_t, o_t))
projection = <g_t, o_t> / (||o_t||^2 + eps) * o_t
correction = -alpha * conflict * projection
g_corrected = g_t + correction
